Makefile

```diff
@@ -0,0 +1,54 @@
+# Copyright (2019) Cobalt Speech and Language Inc.
+
+.PHONY: all deps gen depsbin deps-protoc
+all: gen
+
+SHELL := /bin/bash
+
+TOP := $(shell pwd)
+
+DEPSBIN := ${TOP}/deps/bin
+DEPSGO := ${TOP}/deps/go
+DEPSTMP := ${TOP}/deps/tmp
+$(shell mkdir -p $(DEPSBIN) $(DEPSGO) $(DEPSTMP))
+
+DEPSVENV := ${TOP}/deps/venv
+
+export PATH := ${DEPSBIN}:${DEPSGO}/bin:$(PATH)
+deps: deps-protoc deps-hugo deps-gendoc deps-gengo deps-gengateway deps-py
+
+deps-protoc: ${DEPSBIN}/protoc
+${DEPSBIN}/protoc:
+	cd ${DEPSBIN}/../ && wget \
+		"https://github.com/protocolbuffers/protobuf/releases/download/v3.7.1/protoc-3.7.1-linux-x86_64.zip" && \
+		unzip protoc-3.7.1-linux-x86_64.zip && rm -f protoc-3.7.1-linux-x86_64.zip
+
+deps-hugo: ${DEPSBIN}/hugo
+${DEPSBIN}/hugo:
+	cd ${DEPSBIN} && wget \
+		"https://github.com/gohugoio/hugo/releases/download/v0.55.4/hugo_0.55.4_Linux-64bit.tar.gz" -O - | tar xz hugo
+
+deps-gendoc: ${DEPSBIN}/protoc-gen-doc
+${DEPSBIN}/protoc-gen-doc:
+	cd ${DEPSBIN} && wget \
+		"https://github.com/pseudomuto/protoc-gen-doc/releases/download/v1.3.0/protoc-gen-doc-1.3.0.linux-amd64.go1.11.2.tar.gz" -O - | tar xz --strip-components=1
+
+deps-gengo: ${DEPSGO}/bin/protoc-gen-go
+${DEPSGO}/bin/protoc-gen-go:
+	cd $(DEPSTMP) && mkdir gengo && cd gengo && go mod init tmp && GOPATH=${DEPSGO} go get github.com/golang/protobuf/protoc-gen-go@v1.3.1 && cd .. && rm -rf gengo
```

Comment:
We might want to have a `make clean` target that removes the tmp directory.  If someone encounters an error with `go mod` or `go get`, the temporary files might still exist and prevent it from working the second time through.

Reply:
I'm forcing the cleanup before the mod init and also adding a `make clean` target for removing deps entirely.